### PR TITLE
Updated rds orchestrate to work with both PostGres and MySQL/MariaDB

### PR DESCRIPTION
**Pull request by blarghmatey**

### CODE REVIEW

File: salt/orchestrate/aws/rds.sls [hunk 1]
```diff
@@ -67,16 +67,23 @@ create_{{ ENVIRONMENT }}_{{ dbconfig.name }}_rds_store:
...
-    - description: Backend to create dynamic PostGreSQL credentials for {{ ENVIRONMENT }}
-    - mount_point: postgresql-{{ ENVIRONMENT }}-{{ dbconfig.name }}
+    - backend_type: database
+    - description: Backend to create dynamic {{ dbconfig.engine }} credentials for {{ ENVIRONMENT }}
+    - mount_point: {{ mount_point }}
     - ttl_max: {{ SIX_MONTHS }}
     - ttl_default: {{ SIX_MONTHS }}
     - lease_max: {{ SIX_MONTHS }}
     - lease_default: {{ SIX_MONTHS }}
+    - connection_config_path:
```
Comment (shaidar) [inline salt/orchestrate/aws/rds.sls]: What does this serve?

File: salt/orchestrate/aws/rds.sls [hunk 2]
```diff
@@ -67,16 +67,23 @@ create_{{ ENVIRONMENT }}_{{ dbconfig.name }}_rds_store:
...
     - ttl_default: {{ SIX_MONTHS }}
     - lease_max: {{ SIX_MONTHS }}
     - lease_default: {{ SIX_MONTHS }}
+    - connection_config_path: 
     - connection_config:
+        plugin_name: {{ dbconfig.vault_plugin }}
+        {% if dbconfig.engine == 'postgres' %}
         connection_url: "postgresql://{{ master_user }}:{{ master_pass }}@postgresql-{{ dbconfig.name }}.service.{{ ENVIRONMENT }}.consul:5432/{{ dbconfig.name }}"
+        {% else %}
+        connection_url: "{{ master_user }}:{{ master_pass }}@tcp(mysql.service.{{ ENVIRONMENT }}.consul:3306)/"
```
Comment (shaidar) [inline salt/orchestrate/aws/rds.sls]: Why did you leave out the {{ dbconfig.name }} at the end of this connection_url?

File: salt/orchestrate/aws/security_groups.sls [hunk 3]
```diff
@@ -189,6 +189,42 @@ create_public_postgres_rds_security_group:
...
+          cidr_ip:
+            - {{ VPC_CIDR }}
+    - tags:
+        Name: mysql-rds-{{ VPC_RESOURCE_SUFFIX }}
+        business_unit: {{ BUSINESS_UNIT }}
+        Department: {{ BUSINESS_UNIT }}
+        OU: {{ BUSINESS_UNIT }}
+        Environment: {{ ENVIRONMENT }}
+
+create_public_mysql_rds_security_group:
```
Comment (shaidar) [inline salt/orchestrate/aws/security_groups.sls]: Why is this needed?

File: salt/orchestrate/aws/rds.sls (hunk 2 above)
Comment (blarghmatey) [reply salt/orchestrate/aws/rds.sls]: Because it's not required for MySQL connections

File: salt/orchestrate/aws/rds.sls (hunk 1 above)
Comment (blarghmatey) [reply salt/orchestrate/aws/rds.sls]: That is me forgetting to add the parameter that gets passed to it... I'll fix that.

File: salt/orchestrate/aws/security_groups.sls (hunk 3 above)
Comment (blarghmatey) [reply salt/orchestrate/aws/security_groups.sls]: That is necessary to ensure that the MySQL security groups exist so that the database can be accessed.

Comment (shaidar) [reply salt/orchestrate/aws/security_groups.sls]: Can it not just be access from the VPC as opposed to publicly?

**Comment by shaidar** [comment]
Looks good! Just want to clarify that one last question about the sec_group allowing that cidr_ip range as opposed to maybe the VPC

**Comment by blarghmatey** [comment]
@shaidar the public security group is for cases where we need to allow external access, such as when we are using RDS for heroku apps.

**Comment by shaidar** [comment]
:thumbsup:
